staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3447,7 +3449,16 @@ func (c *Cloud) findELBSubnets(internalELB bool) ([]string, error) {
 			continue
 		}
 
-		// Try to break the tie using a tag
+		// Prefer the one with the cluster Tag
+		existingHasClusterTag := c.tagging.hasClusterTag(existing.Tags)
```

Comment:
i think this logic is incorrect here.
assuming we are matching for public ELB.
we first picked a subnet-1(with no cluster tag but have `elb` role tag). subnet-1 will be "existing"
now we encounter subnet-2(with cluster tag but with `internal-elb` role tag). subnet-2 will chosen with this logic, which is incorrect.

Reply:
this is for backwards compatibility - to prefer subnets with cluster tag over those without.

Reply:
As per our discussion, I will modify the semantics of cluster tag. Having the cluster tag will make the subnet usable only on the tagged cluster, and no cluster tag will make it usable across all clusters in the VPC. In case of more than one usable subnet on the same AZ, we will give preference to the one with the correct role tag.

For example, there are two subnets in the same AZ
subnet-1 (role elb)
subnet-2 (cluster tag, role internal-elb)
For public LB, subnet-1 will be choosen.